src/network/networkd-neighbor.c

```diff
@@ -133,9 +133,15 @@ int neighbor_configure(Neighbor *neighbor, Link *link, link_netlink_message_hand
         if (r < 0)
                 return log_error_errno(r, "Could not set flags: %m");
 
-        r = sd_netlink_message_append_data(req, NDA_LLADDR, &neighbor->mac, sizeof(neighbor->mac));
-        if (r < 0)
-                return log_error_errno(r, "Could not append NDA_LLADDR attribute: %m");
+        if (neighbor->use_mac) {
+                r = sd_netlink_message_append_data(req, NDA_LLADDR, &neighbor->lladdr.mac, sizeof(neighbor->lladdr.mac));
+                if (r < 0)
+                        return log_error_errno(r, "Could not append NDA_LLADDR attribute: %m");
+        } else {
+                r = sd_netlink_message_append_data(req, NDA_LLADDR, &neighbor->lladdr.ip.in, sizeof(neighbor->lladdr.ip.in));
+                if (r < 0)
+                        return log_error_errno(r, "Could not append NDA_LLADDR attribute: %m");
+        }
```

Comment:
you could use the failure check and error message for both if branches. i.e.

```c
if (neighbor->use_mac) 
        r = sd_netlink_message_append_data(…
else
        r = sd_netlink_message_append_data(…
if (r < 0) 
        return …
```

Reply:
Done.